pl_bolts/datasets/array_dataset.py

```diff
@@ -0,0 +1,37 @@
+from typing import Tuple
+
+from pytorch_lightning.utilities import exceptions
+from torch.utils.data import Dataset
+
+from pl_bolts.datasets.base_dataset import ARRAYS, DataModel
+
+
+class ArrayDataset(Dataset):
+    """Dataset wrapping tensors, lists, numpy arrays.
+
+    Attributes:
+        data_models: Sequence of data models.
+
+    Raises:
+        MisconfigurationException: if there is a shape mismatch between arrays in the first dimension.
+    """
+
+    def __init__(self, *data_models: DataModel) -> None:
+        self.data_models = data_models
+
+        if not self._equal_size():
+            raise exceptions.MisconfigurationException("Shape mismatch between arrays in the first dimension")
+
+    def __len__(self) -> int:
+        return len(self.data_models[0].data)
+
+    def __getitem__(self, idx: int) -> Tuple[ARRAYS, ...]:
+        return tuple(data_model.process(data_model.data[idx]) for data_model in self.data_models)
```

Comment:
I don't think we want to call `process` on the data for every `__getitem__` as this could get very expensive very fast

Reply:
I do agree with you. However, from what I have seen every Dataset performs the transform for every `__getitem__` so I was being consistent. What do you recommend going forward?

Reply:
Gah, that's sad. I fully agree, that keeping up with rest of the world probably makes more sense, so consistency over anything else, but I just feel it's so inefficient.

Your call :zap: I'll be fine either way :zap: